docs/guides/compile-contracts.md

```diff
@@ -35,57 +116,52 @@ Each artifact consists of a json with the following properties:
 
 - `deployedLinkReferences`: The deployed bytecode's link references object [as returned by solc](https://solidity.readthedocs.io/en/latest/using-the-compiler.html). If the contract doesn't need to be linked, this value contains an empty object.
 
-## Configuring the compiler
 
-If you need to customize the `solc` compiler options, then you can do so through the `solc` config field in your `hardhat.config.js`, which is an optional object that can contain the following properties:
+The **debug file** has all the information that is necessary to reproduce the compilation and to debug the contracts: this includes the original solc input and output, and the solc version used to compile it.
 
-- `version`: the solc version to use. We recommend always setting this field. Default value: `"0.5.15"`.
+### Build info files
+Hardhat optimizes compilation by compiling the smallest possible set of files at a time. Files that are compiled together have the same solc input and output. Since having this in each debug file would be meaningfully wasteful, this information is deduplicated in build info files that are placed in `artifacts/build-info`. Each contract debug file contains a relative path to its build info file, and each build info file contains the solc input, solc output and the solc version used.
 
-- `optimizer`: an object with `enabled` and `runs` keys. Default value: `{ enabled: false, runs: 200 }`.
+You shouldn't interact with these files directly. 
 
-- `evmVersion`: a string controlling the target evm version. One of `homestead`, `tangerineWhistle`, `spuriousDragon`, `byzantium`, `constantinople`, `petersburg`, and `instanbul`. Default value: managed by `solc`. 
+### Reading artifacts
 
-## Solidity 5 and Solidity 6 contracts in the same project
+The [HRE] has an `artifacts` object with helper methods. For example, you can get a list with the paths to all artifacts by calling `hre.artifacts.getArtifactPaths()`.
 
-Hardhat can handle scenarios where you need to compile Solidity 5 and Solidity 6 contracts in the same project. An example of this is when there are deployed contracts that have been written in Solidity 5 but your new contracts that depend on the old ones are written in Solidity 6, or different contract dependencies use different Solidity versions and you need both of these to play along to run your tests.
+You can also read an artifact using the name of the contract by calling `hre.artifacts.readArtifact("Bar")` and that will give us the content of the artifact for the `Bar` contract. This would only work if there was just one contract `Bar` in the whole project, but calling `hre.artifacts.readArtifact("Foo")`, would yield an error if there were two `Foo` contracts. To disambiguate this case, you would have to use the **Fully Qualified Name** of the contract: `hre.artifacts.readArtifact("contracts/Foo.sol:Foo")`.
```

Comment:
```suggestion
You can also read an artifact using the name of the contract by calling `hre.artifacts.readArtifact("Bar")` and that will give us the content of the artifact for the `Bar` contract. This would only work if there was just one contract `Bar` in the whole project, but calling `hre.artifacts.readArtifact("Foo")`, would throw an error if there were two `Foo` contracts. To disambiguate this case, you would have to use the **Fully Qualified Name** of the contract: `hre.artifacts.readArtifact("contracts/Foo.sol:Foo")`.
```